scripts/build/termux_step_massage.sh

```diff
@@ -60,27 +60,42 @@ termux_step_massage() {
 		fi
 	fi
 
-	local pattern=""
-	for file in ${TERMUX_PKG_NO_SHEBANG_FIX_FILES}; do
-		if [[ -z "${pattern}" ]]; then
-			pattern="${file}"
-			continue
+	if [ "$TERMUX_PKG_NO_SHEBANG_FIX" != "true" ]; then
+		local no_shebang_replace_full_regex="" no_shebang_replace_regex
+		# Assume files are a regex list and special characters in paths in each regex are already escaped.
+		if [[ -n "$TERMUX_PKG_NO_SHEBANG_FIX_FILES" ]]; then
+			while IFS= read -r no_shebang_replace_regex; do
+				[[ -n "$no_shebang_replace_full_regex" ]] && no_shebang_replace_full_regex+='|'
+				no_shebang_replace_full_regex+="($no_shebang_replace_regex)"
+			done < <(printf "%s\n" "$TERMUX_PKG_NO_SHEBANG_FIX_FILES")
+			if [[ -n "$no_shebang_replace_full_regex" ]]; then
+				no_shebang_replace_full_regex='(|./)('"$no_shebang_replace_full_regex"')$'
+			fi
 		fi
-		pattern+='|'"${file}"
-	done
-	if [[ -n "${pattern}" ]]; then
-		pattern='(|./)('"${pattern}"')$'
-	fi
 
-	if [ "$TERMUX_PKG_NO_SHEBANG_FIX" != "true" ]; then
+		local prefix_escaped shebang_already_valid_regex header_line shebang_match
+		local shebang_regex='^#!.*/bin/.*'
+
+		# Escape '\$[](){}|^.?+*' with backslashes.
+		prefix_escaped="$(printf "%s" "$TERMUX_PREFIX_CLASSICAL" | sed -zE -e 's/[][\.|$(){}?+*^]/\\&/g')"
+		shebang_already_valid_regex='^#! ?((/system/)|('"$prefix_escaped"'/))'
+
 		# Fix shebang paths:
-		while IFS= read -r -d '' file; do
-			if [[ -n "${pattern}" ]] && [[ -n "$(echo "${file}" | grep -E "${pattern}")" ]]; then
-				echo "INFO: Skip shebang fix for ${file}"
-				continue
-			fi
-			if head -c 100 "$file" | head -n 1 | grep -E "^#!.*/bin/.*" | grep -q -E -v -e "^#! ?/system" -e "^#! ?$TERMUX_PREFIX_CLASSICAL"; then
-				sed --follow-symlinks -i -E "1 s@^#\!(.*)/bin/(.*)@#\!$TERMUX_PREFIX/bin/\2@" "$file"
+		while IFS= read -rd '' file; do
+			read -rn100 header_line < "$file"
```

Comment:
Use `${#TERMUX_PREFIX_CLASSICAL} + 4` instead of `100` as otherwise regex will not match if prefix length is longer.

Reply:
Done.